sklearn/model_selection/_split.py

```diff
@@ -913,6 +915,238 @@ def get_n_splits(self, X, y, groups):
         return int(comb(len(np.unique(groups)), self.n_groups, exact=True))
 
 
+class _RepeatedSplits(with_metaclass(ABCMeta)):
+    """Repeated splits for K-Fold and Stratified K-Fold
+
+    Repeats splits for cross-validators n times.
+
+    Parameters
+    ----------
+    cv : object, KFold or StratifiedKFold
+        Instance of cross-validator.
+
+    n_repeats : int, default=5
+        Number of times cross-validator needs to be repeated.
+
+    random_state : None, int or RandomState, default=None
+        Random state to be used to generate random state for each
+        repetition.
+    """
+    def __init__(self, cv, n_repeats=5, random_state=None):
+        if not isinstance(cv, (KFold, StratifiedKFold)):
+            raise ValueError(
+                "cv must be an instance of KFold or StratifiedKFold.")
+
+        if not isinstance(n_repeats, (np.integer, numbers.Integral)):
+            raise ValueError("Number of repetitions must be of Integral type.")
+
+        if n_repeats <= 1:
+            raise ValueError("Number of repetitions must be greater than 1.")
+
+        self.cv = cv
+        self.n_repeats = n_repeats
+        self.random_state = random_state
+
+        # make shuffle always true
+        self.cv.shuffle = True
+
+    def split(self, X, y=None, groups=None):
+        """Generates indices to split data into training and test set.
+
+        Parameters
+        ----------
+        X : array-like, shape (n_samples, n_features)
+            Training data, where n_samples is the number of samples
+            and n_features is the number of features.
+
+        y : array-like, of length n_samples
+            The target variable for supervised learning problems.
+
+        groups : array-like, with shape (n_samples,), optional
+            Group labels for the samples used while splitting the dataset into
+            train/test set.
+
+        Returns
+        -------
+        train : ndarray
+            The training set indices for that split.
+
+        test : ndarray
+            The testing set indices for that split.
+        """
+        cv = self.cv
+        n_repeats = self.n_repeats
+        rng = check_random_state(self.random_state)
+
+        for idx in range(n_repeats):
+            cv.random_state = rng.randint(np.iinfo(np.int32).max)
+            for train_index, test_index in cv.split(X, y, groups):
+                yield train_index, test_index
+
+    def get_n_repeats(self):
+        """Returns number of repetitions to be performed"""
+        return self.n_repeats
+
+
+class RepeatedKFold(with_metaclass(ABCMeta)):
```

Comment:
Removing ABC. Not sure what I was thinking when I added this.

Also, I didn't inherit because I thought it made more sense to leave them separate for the case when `_RepeatedSplits` is made public. But inheritance doesn't hurt that case either. So using inheritance now.